### DISCUSSION TITLE
Desklet: photoframe@cinnamon.org: Only Allow Pictures Of A Certain Aspect Ratio

### DISCUSSION THREAD

**Discussion by trymeouteh** [linuxmint/discussions#692] [Ideas]
Have settings for the desklet to only display pictures of a certain aspect ratio to prevent images being stretched in the photo frame.

Settings such as

Aspect Ration : [Any, Portrait, Landscape]

**Comment by lukasz-falda** [comment]
I just fixed the aspect ratio issue with this desklet. I'll be testing it for a while and if possible, I'll submit a PR with the new desklet to https://github.com/linuxmint/cinnamon-spices-desklets.

**Comment by mtwebster** [comment]
Distortion is fixed in Cinnamon 6.6 - I don't think filtering pictures by aspect ratio makes sense - the bug was that the image's aspect ratio wasn't being respected.

https://github.com/linuxmint/cinnamon/commit/3d57bc7cbc9410b9a05a5c43447d690731e1f44f

**Comment by lukasz-falda** [comment]
Great! I see that the sepia configuration has also been fixed.
I did it differently, and also added the “no background” feature.

I agree that the “Aspect ratio: [any, vertical, horizontal]” option is unnecessary.

My desklet doesn't add much more in that case - I'll keep it to myself 😃 

Thanks!
